liblwgeom/lwgeom_geos_node.c

```diff
@@ -146,106 +109,28 @@ lwgeom_node(const LWGEOM* lwgeom_in)
 		return NULL;
 	}
 
-	ep = lwgeom_extract_unique_endpoints(lwgeom_in);
-	if ( ! ep ) {
-		GEOSGeom_destroy(g1);
-		lwerror("Error extracting unique endpoints from input");
-		return NULL;
-	}
-
-	/* Unary union input to fully node */
-	gu = GEOSUnaryUnion(g1);
+	gn = GEOSNode(g1);
 	GEOSGeom_destroy(g1);
-	if ( ! gu ) {
-		lwgeom_free(ep);
-		lwerror("GEOSUnaryUnion: %s", lwgeom_geos_errmsg);
+	if ( ! gn ) {
+		lwerror("GEOSNode: %s", lwgeom_geos_errmsg);
 		return NULL;
 	}
 
 	/* Linemerge (in case of overlaps) */
-	gm = GEOSLineMerge(gu);
-	GEOSGeom_destroy(gu);
+	gm = GEOSLineMerge(gn);
+	GEOSGeom_destroy(gn);
```

Comment:
please check if this is still needed, as GEOSNode might be taking care of line-merging already